File: OBLIVIOUS-VPS/runtime/setup-vps/resources/hub-services/providers/XAIAdapter.js
const BaseAdapter = require("./BaseAdapter");
const { parseTradeDecision } = require("./OpenAIAdapter");

class XAIAdapter extends BaseAdapter {
  constructor(opts) { super({ name: "xai", ...opts }); }

  _buildRequest(prompt) {
    return {
      url: "https://api.x.ai/v1/chat/completions",
      headers: {
        "content-type": "application/json",
        authorization: `Bearer ${this.apiKey}`,
      },
      data: {
        model: "grok-2-latest",
        messages: [
          { role: "system", content: prompt.system },
          { role: "user",   content: prompt.user   },
        ],
        temperature: 0.2,
        max_tokens: 200,
      },
    };
  }

  _parseResponse(data) {
    const text = data?.choices?.[0]?.message?.content || "";
    return parseTradeDecision(text);
  }
}

module.exports = { XAIAdapter };

File: OBLIVIOUS-DEV/project/legacy-frozen/AiMindBridge/AiMindBridge.Server/Providers/XAIAdapter.cs
using System.Text;
using System.Text.Json;
using AiMindBridge.Config;
using Microsoft.Extensions.Logging;

namespace AiMindBridge.Providers;

/// <summary>xAI Grok adapter — uses OpenAI-compatible endpoint.</summary>
public sealed class XAIAdapter : ProviderAdapterBase
{
    public override string Name => "xai";
    private static readonly string DefaultModel = "grok-beta";
    private static readonly string BaseUrl       = "https://api.x.ai/v1/chat/completions";

    public XAIAdapter(ProviderSettings settings, ILogger<XAIAdapter> logger)
        : base(settings, logger) { }

    protected override object BuildHttpPayload(string prompt, string apiKey)
    {
        Http.DefaultRequestHeaders.Clear();
        Http.DefaultRequestHeaders.Add("Authorization", $"Bearer {apiKey}");
        return new
        {
            model       = string.IsNullOrEmpty(Settings.Model) ? DefaultModel : Settings.Model,
            max_tokens  = Settings.MaxTokens,
            temperature = Settings.Temperature,
            messages    = new[]
            {
                new { role = "system", content = "You are a professional forex trade planner. Always respond only with valid JSON." },
                new { role = "user",   content = prompt }
            }
        };
    }

    protected override async Task<string> SendHttpAsync(object payload, CancellationToken ct)
    {
        var url     = string.IsNullOrEmpty(Settings.BaseUrl) ? BaseUrl : Settings.BaseUrl;
        var json    = JsonSerializer.Serialize(payload);
        var content = new StringContent(json, Encoding.UTF8, "application/json");
        var resp    = await Http.PostAsync(url, content, ct);
        resp.EnsureSuccessStatusCode();
        var body = await resp.Content.ReadAsStringAsync(ct);
        using var doc = JsonDocument.Parse(body);
        return doc.RootElement
            .GetProperty("choices")[0]
            .GetProperty("message")
            .GetProperty("content")
            .GetString() ?? "";
    }
}
